ow_plexil/rqt_plexil_plan_selection/src/rqt_plexil_plan_selection/rqt_plexil_plan_selection.py

```diff
@@ -0,0 +1,226 @@
+#The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+#Research and Simulation can be found in README.md in the root directory of
+#this repository.
+
+import os
+import rospy
+import actionlib
+import rospkg
+
+from argparse import ArgumentParser
+from qt_gui.plugin import Plugin
+from python_qt_binding import loadUi
+from python_qt_binding.QtCore import pyqtSignal
+from python_qt_binding.QtGui import QBrush, QColor, QIcon
+from python_qt_binding.QtWidgets import QWidget, QAbstractItemView, QTableWidgetItem, QAbstractScrollArea, QMessageBox, QApplication
+from ow_plexil.msg import PlanSelectionCommand
+from std_msgs.msg import String
+
+class PlexilPlanSelection(Plugin):
+
+  #sets up our signal for the sub callback
+  monitor_signal = pyqtSignal(['QString']) 
+
+  def __init__(self, context):
+    '''init initializes the widget and sets up our subscriber, publisher and event handlers'''
+    super(PlexilPlanSelection, self).__init__(context)
+    self.setObjectName('PlexilPlanSelection')
+
+    # Process standalone plugin command-line arguments
+    parser = ArgumentParser()
+    parser.add_argument("-q", "--quiet", action="store_true", dest="quiet", help="Put plugin in silent mode")
+    args, unknowns = parser.parse_known_args(context.argv())
+
+    # Find resources and Create QWidget
+    self._widget = QWidget()
+    ui_file = os.path.join(rospkg.RosPack().get_path('ow_plexil'), 'rqt_plexil_plan_selection', 'resource', 'plexil_plan_selection.ui')
+    loadUi(ui_file, self._widget)
+    self._widget.setObjectName('PlexilPlanSelectionUI')
+    if context.serial_number() > 1:
+      self._widget.setWindowTitle(self._widget.windowTitle() + (' (%d)' % context.serial_number()))
+    context.add_widget(self._widget)
+
+    #set window icon
+    icon = os.path.join(rospkg.RosPack().get_path('ow_plexil'), 'rqt_plexil_plan_selection', 'resource', 'icon.png')
+    self._widget.setWindowIcon(QIcon(icon))
+    #pub and sub setup
+    self.command_publisher = rospy.Publisher('plexil_plan_selection_commands', PlanSelectionCommand, queue_size=20)
+    self.status_subscriber = rospy.Subscriber('plexil_plan_selection_status', String, self.status_callback)
```

Comment:
I would rather use a service type or ROS Action instead of topics for this type of requests

Reply:
Just to clarify the status message can remain as a topic.